qiskit/providers/models/pulsedefaults.py

```diff
@@ -94,44 +98,335 @@ class Command(BaseModel):
+            *params: Command parameters for generating the output schedule.
+            **kwparams: Keyworded command parameters for generating the schedule.
+
+        Returns:
+            The Schedule defined for the input.
+
+        Raises:
+            PulseError: If the operation is not defined on the qubits.
+        """
+        self.assert_has(operation, qubits)
+        schedule = self._ops_def[operation].get(_to_tuple(qubits))
+        if isinstance(schedule, ParameterizedSchedule):
+            schedule = schedule.bind_parameters(*params, **kwparams)
+        return schedule
+
+    def get_parameters(self, operation: str, qubits: Union[int, Iterable[int]]) -> Tuple[str]:
+        """
+        Return the list of parameters taken by the given operation on the given qubits.
+
+        Args:
+            operation: Name of the operation.
+            qubits: The qubits for the operation.
+
+        Returns:
+            The names of the parameters required by the operation.
+
+        Raises:
+            PulseError: If the operation is not defined on the qubits.
+        """
+        self.assert_has(operation, qubits)
+        return self._ops_def[operation][_to_tuple(qubits)].parameters
+
+    def add(self,
+            operation: str,
+            qubits: Union[int, Iterable[int]],
+            schedule: [Schedule, ParameterizedSchedule]) -> None:
+        """
+        Add a new known operation.
+
+        Args:
+            operation: The name of the operation to add.
+            qubits: The qubits which the operation applies to.
+            schedule: The Schedule that implements the given operation.
+
+        Returns:
+            None
+
+        Raises:
+            PulseError: If the qubits are provided as an empty iterable.
+        """
+        qubits = _to_tuple(qubits)
+        if qubits == ():
+            raise PulseError("Cannot add definition {} with no target qubits.".format(operation))
+        if not isinstance(schedule, (Schedule, ParameterizedSchedule)):
+            raise PulseError("Attemping to add an invalid schedule type.")
+        if self.has(operation, qubits):
+            warnings.warn("Replacing previous definition of {} on qubit{} "
+                          "{}.".format(operation,
+                                       's' if len(qubits) > 1 else '',
+                                       qubits if len(qubits) > 1 else qubits[0]))
+        self._ops_def[operation][qubits] = schedule
+
+    def remove(self, operation: str, qubits: Union[int, Iterable[int]]) -> None:
+        """Remove the given operation from the defined operations.
+
+        Args:
+            operation: The name of the operation to add.
+            qubits: The qubits which the operation applies to.
+
+        Returns:
+            None
+
+        Raises:
+            PulseError: If the operation is not present.
+        """
+        self.assert_has(operation, qubits)
+        self._ops_def[operation].pop(_to_tuple(qubits))
```

Comment:
Also need to check if this operation now no longer has qubits and remove it as well from `_ops_def`. The same for `pop`